### PR TITLE
feat: basic leave game functionality

### PR DESCRIPTION
**Pull request by mikejseay**
This sketches up the basic functionality for game members to leave the game, focused on allowing Editors to cleanly leave the game. Editors see a new accordion menu that shows a button that allows them to leave (it's hidden so they don't accidentally do it).

Since leaving the game is kind of meaningless for the Host, that's not implemented in any way. And since Spectators leaving has no real effect except to remove their name from the Lobby view, I didn't implement that either.

When a member leaves the game, it navigates them to the root view, deletes the device from the device table, leaves the socket room, turns off the listeners.

When an editor leaves the game, it also removes them from the room carefully, handing off any poems in queue to the next editor, and reorganizes the editors in the room.

Also I had to fix a few things to please ESLint given the new rules about alphabetizing.

### CODE REVIEW

**Comment by mikejseay** [comment]
Also note that this does not implement turn timer or AFK detection, although those features would rely on this.

**Comment by mikejseay** [comment]
Also note that this specifically disallows the VIPEditor from leaving the game, because that solves a couple of problems. 1) If the VIPEditor leaves the game from the lobby during game creation, you would need to set the next Editor to become the VIP, which is tricky. 2) If the VIPEditor leaves the game when there are no other editors in the room, that would destroy all of the poems and their ability to be finalized.

File: src/components/GameSnack/index.tsx [hunk 1]
```diff
@@ -48,7 +48,7 @@ const GameSnack = () => {
 
     return (
         <Snackbar
-            anchorOrigin={{ vertical: "bottom", horizontal: "center" }}
+            anchorOrigin={{ horizontal: "center", vertical: "bottom" }}
```
Comment (peterklingelhofer) [inline src/components/GameSnack/index.tsx:51]: Let's multi-line this object.

```suggestion
            anchorOrigin={{
                horizontal: "center",
                vertical: "bottom",
            }}
```

File: src/components/Leave/index.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import Button from "@mui/material/Button";
+import * as React from "react";
+
+import { useSocket } from "../App";
+import Accordion from "@mui/material/Accordion";
+import AccordionSummary from "@mui/material/AccordionSummary";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import { accordionText, accordionTitle, accordionButton } from "../LineInput/styles";
+import Typography from "@mui/material/Typography";
+import AccordionDetails from "@mui/material/AccordionDetails";
```
Comment (peterklingelhofer) [inline src/components/Leave/index.tsx:10]: Let's sort these imports.

```suggestion
import * as React from "react";
import Accordion from "@mui/material/Accordion";
import AccordionSummary from "@mui/material/AccordionSummary";
import Button from "@mui/material/Button";
import ExpandMoreIcon from "@mui/icons-material/ExpandMore";

import Typography from "@mui/material/Typography";
import AccordionDetails from "@mui/material/AccordionDetails";

import { useSocket } from "../App";
import {
    accordionButton,
    accordionText,
    accordionTitle,
} from "../LineInput/styles";
```

File: src/components/LineInput/index.tsx [hunk 3]
```diff
@@ -13,9 +13,9 @@ import "./LineInput.css";
 import {
     activeInput,
     caret,
-    donePoemAccordionText,
-    donePoemAccordionTitle,
-    donePoemButton,
+    accordionText,
+    accordionTitle,
+    accordionButton,
```
Comment (peterklingelhofer) [inline src/components/LineInput/index.tsx:18]: Let's sort these imports

```suggestion
```

File: src/components/LineInput/index.tsx [hunk 4]
```diff
@@ -13,9 +13,9 @@ import "./LineInput.css";
 import {
     activeInput,
     caret,
```
Comment (peterklingelhofer) [inline src/components/LineInput/index.tsx:15]: Let's sort these imports.

```suggestion
    accordionButton,
    accordionText,
    accordionTitle,
    activeInput,
    caret,
```

File: src/components/Settings/index.tsx [hunk 5]
```diff
@@ -1,15 +1,60 @@
+import * as React from "react";
+import Popover from "@mui/material/Popover";
 import SettingsIcon from "@mui/icons-material/Settings";
 import IconButton from "@mui/material/IconButton";
```
Comment (peterklingelhofer) [inline src/components/Settings/index.tsx:4]: Let's sort these imports.

```suggestion
import * as React from "react";
import IconButton from "@mui/material/IconButton";
import Popover from "@mui/material/Popover";
import SettingsIcon from "@mui/icons-material/Settings";
```

File: src/components/Settings/index.tsx [hunk 6]
```diff
@@ -1,15 +1,60 @@
...
+            <Popover
+                id={id}
+                open={isOpen}
+                anchorEl={anchorEl}
+                onClose={handleClose}
+                anchorOrigin={{
+                    horizontal: "left",
+                    vertical: "bottom",
+                }}
+            >
```
Comment (peterklingelhofer) [inline src/components/Settings/index.tsx:46]: Let's sort these props.

```suggestion
            <Popover
                anchorEl={anchorEl}
                anchorOrigin={{
                    horizontal: "left",
                    vertical: "bottom",
                }}
                id={id}
                onClose={handleClose}
                open={isOpen}
            >
```

**Review by peterklingelhofer — COMMENTED** [review]
Looking really good.

**Review by peterklingelhofer — APPROVED** [review]
Tested and reviewed, the changes work as described.

**Pull request merged**